### ISSUE TITLE
Clicking the leave button in the header apparently doesn't always update Rooms list

### ISSUE THREAD

**Issue by matrixbot** [vector-im/vector-web#578]
Created by @ matthew:matrix.org.

**Comment by Kegsay** [comment]
Duplicate of #573 which is blocked on Synapse SYN-573 (matching issue IDs woo!) https://matrix.org/jira/browse/SYN-573
